plugins/orchestrator/src/utils/formState.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  JsonArray,
+  JsonObject,
+  JsonPrimitive,
+  JsonValue,
+} from '@backstage/types';
+
+export function isJsonPrimitive(value: JsonValue): value is JsonPrimitive {
+  return (
+    typeof value === 'number' ||
+    typeof value === 'string' ||
+    typeof value === 'boolean' ||
+    value === null
+  );
+}
+
+export function isJsonObject(value: JsonValue): value is JsonObject {
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+}
+
+export function isJsonArray(value: JsonValue): value is JsonArray {
+  return Array.isArray(value);
+}
+
+export function flattenParametersFromFormState(
+  formState?: Record<string, JsonValue>,
+): Record<string, JsonValue> {
+  if (!formState) {
+    return {};
+  }
+
+  const parameters: Record<string, JsonValue> = {};
+
+  Object.entries(formState).forEach(([key, value]) => {
+    if (value === undefined) {
+      parameters[key] = '';
+    } else if (isJsonPrimitive(value) || isJsonArray(value)) {
```

Comment:
```suggestion
    } else if (!isJsonObject(value) {
```

Reply:
Good catch!

Reply:
Done in https://github.com/caponetto/backstage-plugins/pull/66/commits/f876365ea97d8780e43cb46ef47585062dbe8bba